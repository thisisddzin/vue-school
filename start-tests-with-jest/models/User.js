export default class User {
  constructor(details){
    const { firstname, lastname } = details
    this.firstname = firstname
    this.lastname = lastname
    this.created_at = new Date()
  }

  get name (){
    return `${this.firstname} ${this.lastname}`
  }
}
