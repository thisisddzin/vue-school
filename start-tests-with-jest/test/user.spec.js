import User from '../models/User'

/* Some other tricks
  1. Can use test.only(...) to test just this test in this file when use 'yarn test'
  2. Can use test.skip(...) to test all test from this file except skiped test
  3. Can use 'yarn test --watchAll' to keep running the tests in background for when change the tests file
*/

describe('User.js', () => {
  let user;

  beforeEach(() => {
    user = new User({ firstname: 'Adenilson', lastname: 'Santos' })
  }) // Needed to use beforeEach cause before the snapshot test was added favoriteMovied and that differe of snapshot
  
  test('name method returns full name' , () => {
    expect(user.name).toBe('Adenilson Santos')
    expect(user.created_at).toEqual(expect.any(Date))
  })

  test('should match instance of Date class', () => {
    user.favoriteMovies = [
      {
        title: 'Harry Potter',
        rate: 10
      },
      {
        title: 'Resident Evil',
        rate: 9
      }
    ]
    expect(user.created_at).toBeInstanceOf(Date)
  })

  test('should match with snapshot', () => {
    delete user.created_at
    expect(user).toMatchSnapshot()
  })

})