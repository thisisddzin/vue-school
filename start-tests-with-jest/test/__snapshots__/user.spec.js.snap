// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User.js should match with snapshot 1`] = `
User {
  "firstname": "Adenilson",
  "lastname": "Santos",
}
`;
